src/lib/InstantSearch.js

```diff
@@ -204,6 +210,9 @@ To help you migrate, please refer to the migration guide: https://community.algo
       this.widgets.push(widget);
     });
 
+    // Second part of the fix for #3148
+    if (lastWidget) this.widgets.push(lastWidget);
```

Comment:
this breaks the unit tests, not sure if it also breaks the usage? I think it happens because the order might be changed; but not sure. See: 

https://travis-ci.org/algolia/instantsearch.js/jobs/432952845#L1074

Reply:
Yes I saw that. The 0th widget is moved back in the last which is consistent with the purpose of the PR. I will fix the test.